google/cloud/pubsub/samples/samples.cc

```diff
@@ -780,30 +785,15 @@ void CreateAvroSchema(google::cloud::pubsub::SchemaServiceClient client,
   //! [START pubsub_create_avro_schema] [create-avro-schema]
   namespace pubsub = ::google::cloud::pubsub;
   [](pubsub::SchemaServiceClient client, std::string const& project_id,
-     std::string const& schema_id) {
-    auto constexpr kDefinition = R"js({
-      "type": "record",
-      "name": "State",
-      "namespace": "utilities",
-      "doc": "A list of states in the United States of America.",
-      "fields": [
-        {
-          "name": "name",
-          "type": "string",
-          "doc": "The common name of the state."
-        },
-        {
-          "name": "post_abbr",
-          "type": "string",
-          "doc": "The postal code abbreviation of the state."
-        }
-      ]
-    })js";
+     std::string const& schema_id, std::string const& schema_definition_file) {
+    auto definition = ReadFile(schema_definition_file);
```

Comment:
FYI (no action required): in general we avoid subroutines in sample code.  Somebody reading the code needs to figure out what the subroutine does and that distracts from the sample. These are pretty small, and it is easy to figure out what they do (almost).  But keep this in mind in future samples.

This is actionable: the one thing that is not obvious in this case is the return type.  A customer reading *just* the code snippet will need to wonder what is the return type for `ReadFile()`, and why it needs a `if (!definition)`.

Consider the suggestion to use exceptions and maybe change the code here to:

```cc
std::string const definition = ReadFile(schema_definition_file);
```